pkg/apis/migration/migration.go

```diff
@@ -407,3 +415,129 @@ func (volumeMigration *volumeMigration) registerVolume(ctx context.Context, volu
 	}
 	return volumeID.Id, nil
 }
+
+// cleanupStaleCRDInstances helps in cleaning up stale volume migration CRD instances
+func (volumeMigration *volumeMigration) cleanupStaleCRDInstances(ctx context.Context) {
+	log := logger.GetLogger(ctx)
+	config, err := k8s.GetKubeConfig(ctx)
+	if err != nil {
+		log.Errorf("failed to get kubeconfig. err: %v", err)
+		return
+	}
+	crdGroupClient, err := k8s.NewClientForGroup(ctx, config, CRDGroupName)
+	if err != nil {
+		log.Errorf("Failed to create new client for group %q . Err: %+v", CRDGroupName, err)
+		return
+	}
+	// Clean up routine to cleanup successful CnsRegisterVolume instances
+	err = volumeMigrationInstance.watcher(ctx)
+	if err != nil {
+		log.Error("Failed to watch on config file for changes to VolumeMigrationCRCleanupIntervalInHours. Error: %+v", err)
+		return
+	}
+	go func() {
+		for {
+			ctx, log = logger.GetNewContextWithLogger()
+			log.Infof("Triggering CnsVSphereVolumeMigration cleanup routine")
+			volumeMigrationResourceList := &migrationv1alpha1.CnsVSphereVolumeMigrationList{}
+			err = crdGroupClient.List(ctx, volumeMigrationResourceList)
+			if err != nil {
+				log.Warnf("failed to get CnsVSphereVolumeMigration list from the cluster. Err: %+v", err)
```

Comment:
We should not  continue if we fail to obtain `volumeMigrationResourceList`

Reply:
Why ? If we fail here, then the cleanup CRD routine will exit and we won't be able to delete/cleanup CRDs forever

Reply:
but if you do not have volumeMigrationResourceList with you. How will you delete any CRs?

Reply:
I mean end this instance of a job with `continue`

Reply:
Okay, continue - sounds good to me